encrypted-media/scripts/setmediakeys-again-after-resetting-src.js

```diff
@@ -0,0 +1,79 @@
+function runTest(config, qualifier) {
+    var testname = testnamePrefix( qualifier, config.keysystem )
+                                     + ', setmediakeys again after resetting src';
+
+    var configuration = getSimpleConfigurationForContent( config.content );
+
+    if ( config.initDataType && config.initData ) {
+        configuration.initDataTypes = [ config.initDataType ];
+    }
+
+    async_test (function (test) {
+        var _video = config.video,
+            _access,
+            _mediaKeys,
+            _mediaKeySession,
+            _mediaSource;
+
+        function onFailure(error) {
+            forceTestFailureFromPromise(test, error);
+        }
+
+        function onMessage(event) {
+            config.messagehandler( event.messageType, event.message ).then( function( response ) {
+                _mediaKeySession.update( response ).catch(onFailure).then(function() {
+                    _video.play();
+                });
+            });
+        }
+
+        function onEncrypted(event) {
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest(   config.initData ? config.initDataType : event.initDataType,
+                                                config.initData || event.initData )
+            .catch(onFailure);
+        }
+
+        function playVideoAndWaitForTimeupdate()
+        {
+            _mediaKeySession = _mediaKeys.createSession('temporary');
+            waitForEventAndRunStep('encrypted', _video, onEncrypted, test);
```

Comment:
This adds a second copy of the event handler on the second time `playVideoAndWaitForTimestamp` is called. This results in `generateRequest` being called twice for this new session, which is what is causing the failure.

The event handler should be added just once, perhaps before calling `requestMediaKeySystemAccess`.